doc/visual-programming/source/widgets/data/datasampler.rst

```diff
@@ -14,53 +14,44 @@ Outputs
+   across users, while *stratify sample* mimics the composition of the
    input dataset.
-4. Produce a report.
-5. Press *Sample data* to output the data sample.
+4. Press *Sample Data* to output the data sample.
  
 
 Examples
 --------
 
-First, let's see how the **Data Sampler** works. Let's look at the
-information on the original dataset in the :doc:`Data Info <../data/datainfo>` widget. We see
-there are 24 instances in the data (we used *lenses.tab*). We sampled
-the data with the **Data Sampler** widget and we chose to go with a fixed
-sample size of 5 instances for simplicity. We can observe the sampled
-data in the :doc:`Data Table <../data/datatable>` widget. The second :doc:`Data Table <../data/datatable>` shows the
-remaining 19 instances that weren't in the sample.
+First, let's see how the **Data Sampler** works. We will use the *iris* data from the :doc:`File <../data/file>` widget. We see there are 150 instances in the data. We sampled the data with the **Data Sampler** widget and we chose to go with a fixed sample size of 5 instances for simplicity. We can observe the sampled data in the :doc:`Data Table <../data/datatable>` widget (Data Table (in-sample)). The second :doc:`Data Table <../data/datatable>` (Data Table (out-of-sample)) shows the remaining 145 instances that weren't in the sample. To output the out-of-sample data, double-click the connection between the widgets and rewire the output to *Remaining Data --> Data*.
 
 .. figure:: images/DataSampler-Example1.png 
 
-In the workflow below, we have sampled 10 data instances from the *Iris*
-dataset and sent the original data and the sample to :doc:`Scatter Plot <../visualize/scatterplot>`
-widget for exploratory data analysis. The sampled data instances are plotted
-with filled circles, while the original dataset is represented with
-empty circles.
+Now, we will use the **Data Sampler** to split the data into training and testing part. We are using the *iris* data, which we loaded with the :doc:`File <../data/file>` widget. In **Data Sampler**, we split the data with *Fixed proportion of data*, keeping 70% of data instances in the sample.
+
+Then we connected two outputs to the :doc:`Test & Score <../evaluation/testandscore>` widget, *Data Sample --> Data* and *Remaining Data --> Test Data*. Finally, we added :doc:`Logistic Regression <../model/logisticregression>` as the learner. This runs logistic regression on the Data input and evaluates the results on the Test Data.
+
+.. figure:: images/DataSampler-Example2.png
+
+Over/Undersampling
+------------------
+
+**Data Sampler** can also be used to oversample a minority class or undersample majority class in the data. Let us show an example for oversampling. First, separate the minority class using a :doc:`Logistic Regression <../data/selectrows>` widget. We are using the *iris* data from the :doc:`File <../data/file>` widget. The data set has 150 data instances, 50 of each class. Let us oversample, say, *iris-setosa*.
```

Comment:
I do not understand the
```
First, separate the minority class using a Logistic Regression widget.
```
sentence, the reset seems ok.

Reply:
Uh, that's a bit of a typo. It should be Select Rows. Will fix!